sklearn/neighbors/graph.py

```diff
@@ -101,7 +107,7 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
     else:
         _check_params(X, metric, p, metric_params)
 
-    query = _query_include_self(X, include_self)
+    query = _query_include_self(X._fit_X, include_self, mode)
```

Comment:
Using a private attribute not in the class itself seems like an antipattern. Given the number of times we use `n_samples_fit_` and `_fit_X`, can we make `fit_X_` public?

Reply:
Not sure `_fit_X` is intended to be public, which is why I added `n_samples_fit_`.
Note that this use of `_fit_X` was prior to this PR, I just moved it out of `_query_include_self`.

Reply:
Ah I see, okay, lets leave this. We can adjust this in another PR.